File: test/py/case15_class_member.py
# このファイルは `test/py/case15_class_member.py` のテスト/実装コードです。
# 役割が分かりやすいように、読み手向けの説明コメントを付与しています。
# 変更時は、既存仕様との整合性とテスト結果を必ず確認してください。

class Counter:
    value: int = 0

    def inc(self) -> int:
        Counter.value += 1
        return Counter.value


if __name__ == "__main__":
    c: Counter = Counter()
    c.inc()
    c = Counter()
    print(c.inc())

File: test/cs/case15_class_member.cs
// このファイルは `test/cs/case15_class_member.cs` の変換結果（C#サンプル）です。
// Python入力との対応関係を追いやすくするため、日本語コメントを付与しています。
// 仕様変更時は、対応する Python 側ケースと合わせて更新してください。

using System;

public static class Program
{
    public class Counter
    {
        public static int value = 0;

        public int inc()
        {
            Counter.value = (Counter.value + 1);
            return Counter.value;
        }
    }

    public static void Main(string[] args)
    {
        Counter c = new Counter();
        Console.WriteLine(c.inc());
    }
}
